llvm/test/CodeGen/RISCV/xtheadmemidx.ll

```diff
@@ -1136,10 +1136,9 @@ define i64 @lrd_large_offset(ptr %a, i64 %b) {
 ; RV32XTHEADMEMIDX-NEXT:    slli a1, a1, 3
 ; RV32XTHEADMEMIDX-NEXT:    add a0, a1, a0
 ; RV32XTHEADMEMIDX-NEXT:    lui a1, 23
-; RV32XTHEADMEMIDX-NEXT:    addi a1, a1, 1792
```

Comment:
Just a thought here - imagine we had a variant of this case where only one the computed offsets was out of bounds.  We could still re-associate to put the addi after the add.  This would break the macro fusion (if any) on the LUI/ADDI sequence, but would allow one user to bypass the ADDI even if the other couldn't.